service/matching/pollerHistory.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package matching
+
+import (
+	"time"
+
+	"github.com/uber/cadence/common/cache"
+)
+
+const (
+	pollerHistoryInitSize    = 0
+	pollerHistoryInitMaxSize = 1000
+	pollerHistoryTTL         = 5 * time.Minute
+)
+
+type (
+	pollerIdentity struct {
```

Comment:
the name is confusing. The identity is the identity supplied by worker which could be customized by user to any string. I think the idea here is to guarantee the uniqueness. Probably use some different name here to distinguish that would help understand the code.

Reply:
well, there is a pollerID, which is actually a one time request ID; identity (which is specified in the API) is to some degree too generic......

let us wait for more comment about the names.